lib/mongodb.js

```diff
@@ -705,6 +705,7 @@ MongoDB.prototype.buildWhere = function(model, where) {
     }
 
     var prop = self.getPropertyDefinition(model, propName);
+    var propIsForeignKey = (prop === undefined);
```

Comment:
Surely there must a better way to check if the property is really a foreign key: it could be a non-existing property name or a mistyped one.

But does StrongLoop connectors check for non-declared field names anywhere else or is it left to the appreciation of the database engine ? I bet for the latter.

Reply:
I will not update getPropertyDefinition() and isObjectIDProperty() since this patch does the job, but still it could be a cleaner solution.

Reply:
I don't think this is the way to solve this problem; the property being undefined could simply be an error.
You could use the `getModelDefinition` function to retrieve the model def and then examine the relations collection to see if the property is present in any of them, though, I admit that would be a lot more work.